corpus/convert.py

```diff
@@ -182,3 +183,45 @@ def run(self):
                 segment.orth = line.strip()
 
         c.dump(tk.uncached_path(self.out_corpus))
+
+
+class CorpusToTextDictJob(Job):
+    """
+    Extract the Text from a Bliss corpus to fit a "{key: text}" structure (e.g. for RETURNN)
+    """
+
+    def __init__(self, corpus, segments=None):
+        """
+        :param Path corpus: bliss corpus file
+        :param Path|None segments: a segment file as optional whitelist
+        """
+        self.corpus_path = corpus
+        self.segments_file_path = segments
+
+        self.out_dictionary = self.output_path("text_dictionary.py")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        c = corpus.Corpus()
+        c.load(tk.uncached_path(self.corpus_path))
+
+        dictionary = {}
+
+        segments = None
+        if self.segments_file_path:
+            with uopen(self.segments_file_path) as f:
+                segments = set(line.decode().strip() for line in f.readlines())
+
+        for segment in c.segments():
+            orth = segment.orth.strip()
+            key = segments.fullname()
+            if segments:
+                if key not in segments:
+                    continue
+            dictionary[key] = orth
+
+        dictionary_string = pprint.pformat(dictionary, width=1000)
+        with open(self.out_dictionary.get_path(), "wt") as f:
```

Comment:
`uopen` uses `tk.uncached_path`

Reply:
hm I know what you mean.. out variables should use `get_path`
but I find `uopen` more readable..
not sure what the best policy here is..
out variables should use `get_path` since we know they are a sisyphus Path Object. But still use uopen when we directly use the file?